### PR TITLE
Changes to SmartIntake, need to be confirmed and pushed to master

### PR DESCRIPTION
**Pull request by JoeyBrar**
-Siddhant

### CODE REVIEW

**Comment by JoeyBrar** [comment]
@Harsh-Sinha Please review.  - Nick & Sid

File: src/main/cpp/subsystems/SmartIntake.cpp [hunk 1]
```diff
@@ -59,43 +59,28 @@ void SmartIntake::HandleInput(RobotControlData& input){
...
-            input.intakeInput.goToPseudoStowPos = false;
-            if (input.intakeOutput.intakePos == IntakePos::PSEUDO_STOW)
-            {
-                ++m_IntakeState;
-            }
-        }
-            break;
+        
         default:
+            m_SmartIntakeFlag = false;
```
Comment (Harsh-Sinha) [inline src/main/cpp/subsystems/SmartIntake.cpp]: This should not be in the default case. Make an explicit case for switching over to false. We use the default case to do nothing

File: src/main/cpp/subsystems/SmartIntake.cpp [hunk 2]
```diff
@@ -59,43 +59,28 @@ void SmartIntake::HandleInput(RobotControlData& input){
...
             break;
         }
     }
     else
     {
         switch (m_IntakeState)
         {
-        case 0:
+        case 2:
```
Comment (Harsh-Sinha) [inline src/main/cpp/subsystems/SmartIntake.cpp]: this should start at case 0.

File: src/main/cpp/subsystems/SmartIntake.cpp [hunk 3]
```diff
@@ -59,43 +59,28 @@ void SmartIntake::HandleInput(RobotControlData& input){
...
         {
-                /* code */
+            input.intakeInput.runIntakeIn = true;
```
Comment (Harsh-Sinha) [inline src/main/cpp/subsystems/SmartIntake.cpp]: checking if the laser hit should be in the true case. The if statement should conditionally transfer to the false case

File: src/main/cpp/subsystems/SmartIntake.cpp [hunk 4]
```diff
@@ -130,27 +115,8 @@ void SmartIntake::HandleInput(RobotControlData& input){
...
-        {
-            input.intakeInput.goToPseudoStowPos = false;
-            if (input.intakeOutput.intakePos == IntakePos::PSEUDO_STOW)
-            {
-                ++m_OutTakeState;
-            }
-        }
-            break;
         default:
+            m_SmartOutTakeFlag = false;
```
Comment (Harsh-Sinha) [inline src/main/cpp/subsystems/SmartIntake.cpp]: see associated comment for intake

File: src/main/cpp/subsystems/SmartIntake.cpp [hunk 5]
```diff
@@ -159,15 +125,18 @@ void SmartIntake::HandleInput(RobotControlData& input){
         
         switch (m_OutTakeState)
         {
-        case 0:
+        case 2:
```
Comment (Harsh-Sinha) [inline src/main/cpp/subsystems/SmartIntake.cpp]: see associated comment for intake

File: src/main/cpp/subsystems/SmartIntake.cpp [hunk 6]
```diff
@@ -159,15 +125,18 @@ void SmartIntake::HandleInput(RobotControlData& input){
...
         {
-                /* code */
+            input.intakeInput.runIntakeOut = true;
```
Comment (Harsh-Sinha) [inline src/main/cpp/subsystems/SmartIntake.cpp]: see associated comment for intake

**Review by Harsh-Sinha — CHANGES_REQUESTED** [review]
Overall, good job. A couple bug fixes to be made.

**Pull request merged**